TEDEFO-4546-4547-add-summary-and-navigation-sections

rousso opened the pull request.

rousso committed:
pom: Configure javadoc plugin to exclude specific source files from doclint

rousso committed:
Add support for EFX-2 summary and navigation sections in view templates.

rousso committed:
Allow node context to be assigned to a context variable

rousso committed:
Make the key of a dictionary evaluate relative to its context.

rousso committed:
Add support for hyperlinks in view templates.

bertrand-lorentz commented on src/main/java/eu/europa/ted/efx/interfaces/MarkupGenerator.java:
```diff
@@ -275,16 +338,48 @@ default Markup renderFragmentInvocation(final String name, final PathExpression
      * Given an evaluation context, and some pre-rendered content, this method returns the code that 
      * iterates over the context and repeats the content for each item in the context.
-     * As of version 2.0.0-alpha.4, the method composeFragmentInvocation(String, Set)
+     * As of version 2.0.0-alpha.6, the method composeFragmentInvocation(String, Set)
```
I'm not sure why this comment is modified, if the change it described was done in alpha.4.

rousso replied:
i changed it because it was wrong... it was not deprecated in neither in alpha.4 nor alpha.5 as it has not been released yet :)

bertrand-lorentz commented on src/main/java/eu/europa/ted/efx/interfaces/MarkupGenerator.java:
```diff
@@ -217,7 +276,7 @@ public interface MarkupGenerator {
      * @return              the code that encapsulates the fragment in the target template.
      */
-    @Deprecated(since = "2.0.0-alpha.4", forRemoval = true)
+    @Deprecated(since = "2.0.0-alpha.6", forRemoval = true)
```
I don't understand why the "deprecated since" version is changed here.

rousso replied:
i changed it because it was wrong... it was not deprecated in neither in alpha.4 nor alpha.5 as it has not been released yet :)

bertrand-lorentz committed:
pom: Update eforms-core dependency to released version

bertrand-lorentz committed:
MarkupGeneratorMock: Refactor composeOutputFile method for readability

bertrand-lorentz committed:
EfxTemplateTranslatorV2Test: Reformat template lines for consistency

Format the templates and expected results consistently across the file,
to make them more readable.

bertrand-lorentz committed:
TranslatorContext: Put field modifiers in the usual order

bertrand-lorentz committed:
NodePathExpression: Fix code formatting

bertrand-lorentz reviewed (approved):
Please review my comments about the changes on deprecation version.
I've added some "cosmetic" commits to improve code formatting, and a small refactoring to extract a method in MarkupGeneratorMock.
